testdata/nil-len.go

```diff
@@ -35,4 +35,6 @@ func fn() {
 
 	if s != nil && len(s) == 0 { // nil check is not redundant here
 	}
+	if s != nil && len(s) != 1 { // nil check is not redundant here
```

Comment:
As usual, I am easy to get confused ;)

When `s = nil`:
- `s != nil && len(s) != 1` is `false`
- `len(s) != 1` is `true`

So Dominik is right.